src/libsync/account.cpp

```diff
@@ -216,6 +216,7 @@ void Account::trySetupPushNotifications()
                 qCInfo(lcAccount) << "Delete push notifications object because authentication failed or connection lost";
                 _pushNotifications->deleteLater();
                 _pushNotifications = nullptr;
+                emit pushNotificationsDisabled(this);
```

Comment:
If you want you could add a test for that signal in `test/testpushnotifications.cpp`.